simple/stats/runner.py

```diff
@@ -261,100 +270,111 @@ def _vertical_specs_dcids(self,
-          fh = input_handler.make_file(input_file)
-          if not self._maybe_set_special_fh(fh):
-            input_mcf_fhs.append(fh)
-        for input_file in sorted(input_handler.list_files(extension=".json")):
-          fh = input_handler.make_file(input_file)
-          self._maybe_set_special_fh(fh)
-
-      self.reporter.report_started(import_files=list(
-          map(lambda fh: fh.basename(), input_fhs + input_mcf_fhs)))
-      for input_fh in input_fhs:
-        self._run_single_import(input_fh)
-      for input_mcf_fh in input_mcf_fhs:
-        self._run_single_mcf_import(input_mcf_fh)
-
-  def _run_single_import(self, input_fh: FileHandler):
-    logging.info("Importing file: %s", input_fh.basename())
-    self._create_importer(input_fh).do_import()
-
-  def _run_single_mcf_import(self, input_mcf_fh: FileHandler):
-    logging.info("Importing MCF file: %s", input_mcf_fh.basename())
-    self._create_mcf_importer(input_mcf_fh, self.output_dir_fh,
+        input_files.append(input_store.as_file())
+
+    for input_file in input_files:
+      self._check_if_special_file(input_file)
+      if match(input_file, "*.csv"):
+        input_csv_files.append(input_file)
+      if match(input_file, "*.mcf"):
+        input_mcf_files.append(input_file)
+
+    # Sort input files alphabetically.
+    input_csv_files.sort(key=lambda f: f.name())
+    input_mcf_files.sort(key=lambda f: f.name())
+
+    self.reporter.report_started(import_files=list(input_csv_files +
+                                                   input_mcf_files))
+    for input_csv_file in input_csv_files:
+      self._run_single_import(input_csv_file)
+    for input_mcf_file in input_mcf_files:
+      self._run_single_mcf_import(input_mcf_file)
+
+  def _run_single_import(self, input_file: File):
+    logging.info("Importing file: %s", input_file.name())
+    self._create_importer(input_file).do_import()
+
+  def _run_single_mcf_import(self, input_mcf_file: File):
+    logging.info("Importing MCF file: %s", input_mcf_file.name())
+    self._create_mcf_importer(input_mcf_file, self.output_dir,
                               self.mode == RunMode.MAIN_DC).do_import()
 
-  def _create_mcf_importer(self, input_fh: FileHandler,
-                           output_dir_fh: FileHandler,
+  def _create_mcf_importer(self, input_file: File, output_dir: Dir,
                            is_main_dc: bool) -> Importer:
-    mcf_file_name = input_fh.basename()
-    output_fh = output_dir_fh.make_file(mcf_file_name)
-    reporter = self.reporter.import_file(mcf_file_name)
-    return McfImporter(input_fh=input_fh,
-                       output_fh=output_fh,
+    # Right now, this overwrites any file with the same name,
+    # so if different input sources have files with the same relative path,
+    # they will clobber each others output. Treating this as an edge case
+    # for now but could resolve by allowing input sources to be mapped to output
+    # locations.
```

Comment:
Perhaps add a comment that this only applies to main dc so ok to fixing it later.

Reply:
One idea is to use the sanitized file path approach that you have used for debug files.

Reply:
Updated comment